pkg/standalone-utils/contracts/BaseRelayer.sol

```diff
@@ -0,0 +1,98 @@
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/Address.sol";
+
+import "@balancer-labs/v2-vault/contracts/interfaces/IVault.sol";
+
+/**
+ * @title BaseRelayer
+ * @notice Allows users to atomically approve a relayer and call multiple actions on it
+ */
+contract BaseRelayer {
+    using Address for address payable;
+
+    IVault private immutable _vault;
+
+    constructor(IVault vault) {
+        _vault = vault;
+    }
+
+    receive() external payable {
+        // Accept ETH transfers only coming from the Vault. This is only expected to happen when joining a pool,
+        // performing a swap or managing a user's balance does not use the full amount of ETH provided.
+        // Any remaining ETH value will be transferred back to this contract and forwarded back to the original sender.
+        _require(msg.sender == address(_vault), Errors.ETH_TRANSFER);
+    }
+
+    function getVault() public view returns (IVault) {
+        return _vault;
+    }
+
+    function setRelayerApproval(bool approved, bytes calldata authorisation) external payable {
+        bytes memory data = abi.encodePacked(
+            abi.encodeWithSelector(_vault.setRelayerApproval.selector, msg.sender, address(this), approved),
+            authorisation
+        );
+        _vaultAction(0, data);
+    }
+
+    /**
+     * @notice Allows calling an arbitrary function on the Vault
+     * @dev To be used only to set relayer approval - other actions should be called with a permanent approval set
+     */
+    function _vaultAction(uint256 value, bytes memory data) private returns (bytes memory) {
+        // solhint-disable-next-line avoid-low-level-calls
+        (bool success, bytes memory result) = address(_vault).call{ value: value }(data);
+
+        // Pass up revert if the call failed
+        if (!success) {
+            // solhint-disable-next-line no-inline-assembly
+            assembly {
+                returndatacopy(0, 0, returndatasize())
+                revert(0, returndatasize())
+            }
+        }
+
+        return result;
+    }
+
+    function multicall(bytes[] calldata data) external payable returns (bytes[] memory results) {
+        results = new bytes[](data.length);
+        for (uint256 i = 0; i < data.length; i++) {
+            // solhint-disable-next-line avoid-low-level-calls
+            (bool success, bytes memory result) = address(this).delegatecall(data[i]);
+
+            if (!success) {
+                // Next 6 lines from https://ethereum.stackexchange.com/a/83577
+                if (result.length < 68) revert("MULTICALL_FAILED");
+                // solhint-disable-next-line no-inline-assembly
+                assembly {
+                    returndatacopy(0, 0, returndatasize())
+                    revert(0, returndatasize())
+                }
```

Comment:
```suggestion
                // If there's no revert reason, provide our own
                if (result.length < 68) revert("MULTICALL_FAILED");
                
                // Otherwise, bubble-up the original one
                // solhint-disable-next-line no-inline-assembly
                assembly {
                    returndatacopy(0, 0, returndatasize())
                    revert(0, returndatasize())
                }
```

Reply:
:man_facepalming: Yeah, I should have spotted that comment needed to be changed.